src/main/java/de/dhbw/karlsruhe/domain/services/dialogs/LeaderboardDialogService.java

```diff
@@ -1,17 +1,168 @@
 package de.dhbw.karlsruhe.domain.services.dialogs;
 
+import de.dhbw.karlsruhe.adapters.cli.output.CliOutputPort;
+import de.dhbw.karlsruhe.domain.models.*;
+import de.dhbw.karlsruhe.domain.ports.dialogs.input.InputPort;
 import de.dhbw.karlsruhe.domain.ports.dialogs.output.LeaderboardOutputPort;
+import de.dhbw.karlsruhe.domain.ports.persistence.LeaderboardStorePort;
 import de.dhbw.karlsruhe.domain.services.DependencyFactory;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 public class LeaderboardDialogService {
 
-  private final LeaderboardOutputPort outputPort;
+    private final LeaderboardOutputPort outputPort;
+    private final InputPort inputPort;
+    private LeaderboardStorePort leaderboardStorePort;
+
+    public LeaderboardDialogService() {
+        this.outputPort = DependencyFactory.getInstance().getDependency(LeaderboardOutputPort.class);
+        this.inputPort = DependencyFactory.getInstance().getDependency(InputPort.class);
+        this.leaderboardStorePort = DependencyFactory.getInstance().getDependency(LeaderboardStorePort.class);
+    }
+
+    public void startLeaderboardDialog() {
+        outputPort.displayLeaderboardOptions();
+
+        int userInput = this.awaitUserInput();
+
+        this.loadAndDisplayCorrectLeaderboard(userInput);
+    }
+
+    public int calculateCompleteLeaderboardScore(boolean isCorrect, long timeInMillis, String difficultyAsString) {
```

Comment:
Wieso kalkuliert der DialogService den Score? wäre es nicht sinnvoll den in einer separaten ScoreCalculation Klasse zu berechnen?